app/src/main/java/com/nearbynest/app/ui/list/PropertyListViewModel.kt

```diff
@@ -15,6 +15,61 @@
  */
 package com.nearbynest.app.ui.list
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.nearbynest.app.BuildConfig
+import com.nearbynest.app.data.remote.response.ImageInfo
+import com.nearbynest.app.data.repository.ImageRepository
+import com.nearbynest.app.di.IoDispatcher
+import com.nearbynest.app.di.MainDispatcher
 import com.nearbynest.app.ui.base.BaseViewModel
+import com.nearbynest.app.utils.extension.onError
+import com.nearbynest.app.utils.extension.onException
+import com.nearbynest.app.utils.extension.onSuccess
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+import com.nearbynest.app.utils.result.Event
 
-class PropertyListViewModel : BaseViewModel()
+@HiltViewModel
+class PropertyListViewModel @Inject constructor(
+    private val imageRepository: ImageRepository,
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
+    @MainDispatcher private val mainDispatcher: CoroutineDispatcher
+) : BaseViewModel() {
+    private val _showShimmer = MutableLiveData<Boolean>(false)
+    val showShimmer: LiveData<Boolean>
+        get() = _showShimmer
+
+    private val _imageList = MutableLiveData<Event<List<ImageInfo>>>()
+    val imageList: LiveData<Event<List<ImageInfo>>>
+        get() = _imageList
+
+    init {
+        getImages()
+    }
+
+    private fun getImages() {
+        viewModelScope.launch(ioDispatcher) {
+            imageRepository.getImages(BuildConfig.API_KEY).onSuccess { result ->
+                withContext(mainDispatcher) {
+                    result.hits?.let { _imageList.value = Event(it) }
+                }
+            }.onError { code, message ->
+                Timber.e(code.toString() + message)
+                withContext(mainDispatcher) {
+                    // Tobe implemented
```

Comment:
Use TODO comments instead

Reply:
It was forbidden in hook, now changed the `detekt.yml` file to allow TODOs.
Previously showing error: This comment contains 'TODO:' that has been defined as forbidden in detekt.